### ISSUE TITLE
Support - embed.ly - for all weblinks [oEmbed]

### ISSUE THREAD

**Issue by 4digitalworld** [Kunena/Kunena-Forum#3769]
#### Steps to reproduce the issue

put on forum post any link from any website or media services.
#### Expected result

link with card - picture+description or right media embed.

<img width="325" alt="cards" src="https://cloud.githubusercontent.com/assets/3662519/14315843/52f0107a-fc08-11e5-9cff-cd8c4006d8d5.png">
#### Actual result

link on website without card, without preview.
#### System information (as much as possible)

Joomla version: 3.5
Kunena version: 4.0.10 or 5.0
Php version: 7.0
Database version: 5.5
#### Additional comments

Please add this integration for publication any website link
http://docs.embed.ly/docs/embedly-api

**Comment by xillibit** [comment]
Hello,

Kunena doesn't support that actually, in kunena 5.0 you will have the possibility to add soudcloud item in Kunena message, but we need to add auto-embded support. Spotify isn't supported actually.

**Comment by 4digitalworld** [comment]
i mean, you can add API from service "[Embed.ly](http://embed.ly/)" to Kunena forum.

and Kunena can show more services like SoundCloud, Spotify, Deezer, YouTube and many more, include cards for weblinks (image+description) - this is like twitter cards or facebook, but for Kunena forum.

for example, JomSocial already use service Embed.ly and no worries about add auto-embded every service.

**Comment by 4digitalworld** [comment]
with [Embed.ly](http://embed.ly/) you can post link, for example - from CNN.com with image preview+description or from SoundCloud (and see media player in forum post).

http://embed.ly/ is free service with API for embed on site.

**Comment by xillibit** [comment]
We need to think to use Embed.ly, for tweet and ebay we use our own intergration things

**Comment by 4digitalworld** [comment]
i know about Twitter and Ebay, but this is bettur solution for forum at this moment and used by many publishers, and also used by JomSocial component.

i think this is no problem for replace all twitter/ebay or only twitter thing on forum (forum settings) for administrators.

this is good solution for future and no more questions like - "when you add SoundCloud, when you add Spotify, when you add cards for websites, etc..."

**Comment by xillibit** [comment]
yes, you are right it's a solution more universal but it's a bit short to implement in Kunena 5.0, so it will goes into 5.1

**Comment by 4digitalworld** [comment]
this is relly good news! can't wait! 5.1!

**Comment by Ruud68** [comment]
Hi, really love the embedding part and what embed.ly is doing, but....
Is embedly open source?
How do they make money with my website when implemented?
What are they doing with all the clicks they collect from my users?
Do I have to inform my users when entering my website that I am using their service (As I am obliged by law to do for e.g. google analytics)?
Will external calls be made to embed.ly 's website?
What will that do for my websites performance?
What will happen if embed.ly has a change of mind and starts charging for their services?

just some thoughts :)
Does the Kunena team have a policy in place for implementing third-party api's?

**Comment by 4digitalworld** [comment]
@Ruud68 

what the difference, if YouTube embed player show adsence or service Embed.ly??? what the difference? you integrate third-party service on your site anyway.

SoundCloud also have paid subscription recently in the United States for listen all legally catalog of music.

This is personally your problem - add any third-party service on your site or no. Set "ON" or "OFF" in Kunena settings for Embed.ly - this is your choice.

Embed.ly is free for 5.000 urls/month. i think your users not put on forum so many links every month. But, if you have so BIGGEST community with users post around 50.000 url on forum every month, i think this is no problem paid $19 for this service - http://embed.ly/pricing

@Ruud68 next time, before asked stupid questions, please learn service.

I'm no have so many time, for waiting, when Kunena add every service i want. Kunena can add Embed.ly and this problem with links and media players can solved. JomSocial and many others do it also.

**Comment by 4digitalworld** [comment]
@Ruud68 anytime you can disable Embed.ly in Kunena settings, like other 3rd party service like YouTube player, SoundCloud, etc.

**Comment by Ruud68** [comment]
Just for the record @sanek4life there are no stupic questions, only stupic answers :) At least that is what I have been brought up with.

If I had done some more investigation then I would have probably known that there is a switch in Kunena to enable / disable this feature.

It could also be integrated as a plugin (perhaps a generic Joomla plugin) so the site owner has the option to install yes / no.

And again, no offense: just my thoughts :)

**Comment by 4digitalworld** [comment]
@810 I wrote more about this here: [https://github.com/joomla/joomla-cms/issues/27385](https://github.com/joomla/joomla-cms/issues/27385)

I think it's possible to add this for a forum, like a whitelist of websites.
